libsolc/libsolc.h

```diff
@@ -57,20 +60,34 @@ char const* solidity_license() SOLC_NOEXCEPT;
 /// The pointer returned must not be freed by the caller.
 char const* solidity_version() SOLC_NOEXCEPT;
 
+/// Allocates a given chunk of memory with @p size bytes.
+///
+/// Use this function inside callbacks to allocate data that is to be passed to the compiler stack,
+/// and use solidity_deallocate() to free this memory again, or solidity_reset() to free it all.
+char* solidity_alloc(size_t _size) SOLC_NOEXCEPT;
+
+/// Explicitly frees the memory that was being allocated with solidity_alloc(size_t size);
+///
+/// Important, this call will abort() in case of any invalid argument being passed to this call.
+void solidity_free(char* _data) SOLC_NOEXCEPT;
+
 /// Takes a "Standard Input JSON" and an optional callback (can be set to null). Returns
 /// a "Standard Output JSON". Both are to be UTF-8 encoded.
 ///
 /// @param _input The input JSON to process.
-/// @param _readCallback The optional callback pointer. Can be NULL.
+/// @param _readCallback The optional callback pointer. Can be NULL, but if not NULL, it then will
```

Comment:
```suggestion
/// @param _readCallback The optional callback pointer. Can be NULL, but if not NULL, it will be called by the compiler to request additional input. Please see the documentation of the type for details.
```